versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/RocksDBStore.java

```diff
@@ -189,19 +189,9 @@ public void load(LoadStep loadstep) throws NotFoundException {
     final ColumnFamilyHandle columnFamilyHandle = getColumnFamilyHandle(saveOp.getType());
 
     try (final Transaction transaction = rocksDB.beginTransaction(new WriteOptions())) {
-      if (condition.isPresent()) {
-        final byte[] buffer = transaction.getForUpdate(new ReadOptions(), columnFamilyHandle, saveOp.getId().toBytes(), true);
-        if (null == buffer) {
-          throw new NotFoundException("Unable to load item with ID: " + saveOp.getId());
-        }
-        final RocksBaseValue<C> consumer = RocksSerDe.getConsumer(saveOp.getType());
-        RocksSerDe.deserializeToConsumer(saveOp.getType(), buffer, consumer);
-
-        if (!consumer.evaluate(translate(condition.get()))) {
-          throw new ConditionFailedException("Condition failed during put operation");
-        }
+      if (!isConditionExpressionValid(transaction, columnFamilyHandle, saveOp.getId(), saveOp.getType(), condition, "load")) {
```

Comment:
I might suggest you change this method name to something like evaluateExpression(), since validity could be interpreted differently.

Reply:
Will address in the next PR/Commit.